internal/app/user-api/controller/user_controller_integration_test.go

```diff
@@ -0,0 +1,182 @@
+		repo.SaveUser(&user)
+
+		url := fmt.Sprintf("/find/%s", user.ID.String())
+		req := httptest.NewRequest(http.MethodGet, url, nil)
+		rec := httptest.NewRecorder()
+		e.ServeHTTP(rec, req)
+
+		assert.Equal(t, http.StatusOK, rec.Code)
+
+		var responseUser model.User
+		err := json.Unmarshal(rec.Body.Bytes(), &responseUser)
+		if err != nil {
+			t.Fatalf("failed to unmarshal response: %v", err)
+		}
+		assert.Equal(t, user.FirstName, responseUser.FirstName)
+		assert.Equal(t, user.LastName, responseUser.LastName)
+		assert.Equal(t, user.Email, responseUser.Email)
+		assert.Equal(t, user.Age, responseUser.Age)
+		t.Cleanup(func() {
+			testCleanUp(db)
+		})
+	})
+
+	t.Run("Add test user 1", func(t *testing.T) {
+		e, db, controller, _ := setupServerAndDB(t)
+
+		userOne := model.User{
+			FirstName: "WexFirstName",
+			LastName:  "WexLastName",
+			Email:     "wexfirstname.wexlastname@wexinc.com",
+			Age:       18,
+		}
+		requestBody, err := json.Marshal(userOne)
+		if err != nil {
+			t.Fatalf("Error marshaling JSON: %v", err)
+		}
+		req := httptest.NewRequest(http.MethodPost, "/save", bytes.NewReader(requestBody))
+		headers := make(http.Header)
+		headers.Set("Content-Type", "application/json")
+		req.Header = headers
+
+		rec := httptest.NewRecorder()
+		c := e.NewContext(req, rec)
+
+		controller.SaveUser(c)
+		assert.Equal(t, http.StatusOK, rec.Code)
+
+		var responseUser model.User
+		error := json.Unmarshal(rec.Body.Bytes(), &responseUser)
+		if error != nil {
+			t.Fatalf("failed to unmarshal error response: %v", err)
+		}
+		assert.Equal(t, userOne.FirstName, responseUser.FirstName)
+		t.Cleanup(func() {
+			testCleanUp(db)
+		})
+	})
+
+	t.Run("Add invalid test user 2", func(t *testing.T) {
+		e, db, controller, _ := setupServerAndDB(t)
+
+		userOne := model.User{
+			FirstName: "WexFirst2Name",
+			LastName:  "WexLast2Name",
+			Email:     "wexfirstname.wexlastnamewexinc.com", // No @ symbol in email
+			Age:       16,                                   // Below minimum age
+		}
+		requestBody, err := json.Marshal(userOne)
+		if err != nil {
+			t.Fatalf("Error marshaling JSON: %v", err)
+		}
+		req := httptest.NewRequest(http.MethodPost, "/save", bytes.NewReader(requestBody))
+		headers := make(http.Header)
+		headers.Set("Content-Type", "application/json")
+		req.Header = headers
+
+		rec := httptest.NewRecorder()
+		c := e.NewContext(req, rec)
+
+		controller.SaveUser(c)
+		assert.Equal(t, http.StatusBadRequest, rec.Code)
+
+		var errorMessage errormessage.ErrorMessage
+		error := json.Unmarshal(rec.Body.Bytes(), &errorMessage)
+		if error != nil {
+			t.Fatalf("failed to unmarshal error response: %v", err)
+		}
+		assert.Equal(t, "Age must be at least 18, Email is not valid", errorMessage.ErrorMessageText)
+		assert.Equal(t, http.StatusBadRequest, errorMessage.ErrorStatus)
+		t.Cleanup(func() {
+			testCleanUp(db)
+		})
+	})
```

Comment:
FWIW I think a use case of a user already exist error would be worthwhile for an integration test.

Reply:
Yep, that's added.